pvlib/irradiance.py

```diff
@@ -931,9 +955,10 @@ def reindl(surface_tilt, surface_azimuth, dhi, dni, ghi, dni_extra,
     '''
 
     cos_tt = aoi_projection(surface_tilt, surface_azimuth,
-                            solar_zenith, solar_azimuth)
+                            solar_zenith, solar_azimuth,
+                            projection_minimum=0)
 
-    cos_solar_zenith = tools.cosd(solar_zenith)
+    cos_solar_zenith = np.maximum(tools.cosd(solar_zenith), 0.01745)
```

Comment:
For Rb this limit might be good, but for HB further below the limit should be zero (already enforced).

Reply:
Agreed. I'll fix this.

Reply:
IMO it is more clear to leave the `cos_tt` and `cos_solar_zenith` calculations alone, and apply the filters to the value of `Rb`.

Reply:
@cwhanse I figured you'd be in favor of being consistent with the matlab approach. I like it because it addresses both #432 and #526.